bibliopixel/drivers/visualizer.py

```diff
@@ -38,14 +33,15 @@ def __init__(self, num=0, width=0, height=0, pixelSize=15, port=1618, stayTop=Fa
             if "windows" in operating_system:
                 exe_string = "start python"
             elif "darwin" in operating_system:
-                exe_string = "python"
+                exe_string = "python2.7"
                 suffix = "&"
             else:
-                exe_string = "python"
+                exe_string = "python2.7"
```

Comment:
I had a visualizer written in Juce before.  That was extremely fast and
cross-platform.

But BIG can of worms to reopen!

On Mon, Jun 20, 2016 at 5:59 PM, adammhaile <notifications@github.com>
wrote:

> In bibliopixel/drivers/visualizer.py
> <https://github.com/ManiacalLabs/BiblioPixel/pull/53#discussion_r67774865>
> :
>
> >                  suffix = "&"
> >              else:
> > -                exe_string = "python"
> > +                exe_string = "python2.7"
>
> You have no idea.... so, the reason it's like that is because I am using
> tkinter for the Visualizer UI which is freaking picky about what
> thread/process the UI runs on and I never got it to run with
> multiprocessing right. So I decided not to waste more time and just spawned
> the process. However, I would rather get a new visualizer than rework this
> one. No one really uses it to my knoweldge. Not even me. I have a secret
> implementation I build in C#/WPF (so Windows only.. I know, I know) that is
> an order of magnitude faster. But I would LOVE any suggestions for
> something better that's NOT in tkinter!
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ManiacalLabs/BiblioPixel/pull/53/files/f1413a5dbdcca6ea516504330b5eeec68eb7d502#r67774865>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAPdsqRkEARvSEuwAia_TiZG1inBrJvOks5qNw1YgaJpZM4I6Iga>
> .
>



-- 
     /t

http://radio.swirly.com - art music radio 24/7 366/1000